step-generation/src/utils/constructInvariantContextFromRunCommands.ts

```diff
@@ -30,47 +32,60 @@ export function constructInvariantContextFromRunCommands(
         const { result, params } = command
         const amount = params.quantity
 
+        const newStagingAreaEntities: StagingAreaEntities =
+          createStagingAreaForInvariantContext(params)
+        let newLabwareEntities: LabwareEntities = {}
+
         if (
           result.definition != null &&
           result.definition.schemaVersion === 2
         ) {
           const def = result.definition
           const labwareDefURI = getLabwareDefURI(def)
 
-          const stackLabwareEntities = result.labwareIds
-            .slice(0, amount)
-            .reduce(
-              (entities: LabwareEntities, labwareId) => ({
-                ...entities,
-                [labwareId]: {
-                  id: labwareId,
-                  labwareDefURI,
-                  def,
-                  pythonName: 'n/a',
-                },
-              }),
-              acc.labwareEntities
-            )
+          newLabwareEntities = result.labwareIds.slice(0, amount).reduce(
+            (entities: LabwareEntities, labwareId) => ({
+              ...entities,
+              [labwareId]: {
+                id: labwareId,
+                labwareDefURI,
+                def,
+                pythonName: 'n/a',
+              },
+            }),
+            {}
+          )
+        }
```

Comment:
it doesn't look like newLabwareEntities is being assigned a value multiple times.

```suggestion
const newLabwareEntities: LabwareEntities =
  result.definition != null && result.definition.schemaVersion === 2
    ? (() => {
        const def = result.definition
        const labwareDefURI = getLabwareDefURI(def)
        return result.labwareIds.slice(0, amount).reduce(
          (entities: LabwareEntities, labwareId) => ({
            ...entities,
            [labwareId]: {
              id: labwareId,
              labwareDefURI,
              def,
              pythonName: 'n/a',
            },
          }),
          {}
        )
      })()
    : {}
```

Reply:
hmmm this does work but i find it more confusing than what i had before haha. i'll add it in though